src/MatrixTwitter.js

```diff
@@ -0,0 +1,987 @@
+    for(var media of tweet.entities.media){
+      if(media.type != 'photo'){
+        continue;
+      }
+      media_promises.push(
+        util.uploadContentFromUrl(
+          this._bridge,
+          media.media_url_https,
+          this._get_intent(tweet.id_str)
+        ).then( (mxc_url) => {
+          return {
+            userId:muser,
+            roomId:roomid,
+            time:time,
+            type:"m.room.message",
+            content:{
+              body: media.display_url,
+              msgtype: "m.image",
+              url: mxc_url
+            }
+          }
+        })
+      );
+    }
+  }
+
+  return Promise.all(media_promises).then(msgs =>{
+    msgs.unshift(newmsg);
+    for(var m in this.msg_queue){
+      if(newmsg.time > this.msg_queue[m].time){
+        this.msg_queue.splice(m,0,msgs);
+        return;
+      }
+    }
+
+    this.msg_queue.push(msgs);
+  }).catch(reason =>{
+    log.error("Twitter","Failed to submit tweet to queue, reason: %s",reason);
+  });
+}
+
+/*
+
+  roomid -
+  depth - The maximum depth of the tweet chain (replies to replies) to be traversed.
+  Set this to how deep you wish to traverse and it will be decreased when the
+  function calls itself.
+*/
+
+
+/**
+ * MatrixTwitter.prototype.process_tweet - Process a given tweet (including
+ * resolving any parent tweets), and submit it to the given room. This function
+ * is recursive, limited to the depth set.
+ *
+ * @param  {type} roomid Matrix Room ID of the room that we are processing.
+ * @param  {type} tweet  description
+ * @param  {type} depth  The maximum depth of the tweet chain (replies to
+ * replies) to be traversed. Set this to how deep you wish to traverse and it
+ * will be decreased when the function calls itself.
+ * @return {Promise[]]}  A promise that resolves once the tweet has been queued.
+ */
+MatrixTwitter.prototype.process_tweet = function(roomid, tweet, depth) {
+    depth--;
+    var muser = "@twitter_" + tweet.user.id_str + ":" + this._bridge.opts.domain;
+    var intent = this._bridge.getIntent(muser);
+
+    var type = "m.text";
+    if (tweet.in_reply_to_status_id_str != null) {
+        type = "m.notice"; // A nicer way to show previous tweets
+    }
+    //log.info("Twitter","Processing tweet:",tweet.text);
+
+    return new Promise( (resolve) => {
+      if (tweet.in_reply_to_status_id_str != null && depth > 0) {
+          this.app_twitter.get('statuses/show/' + tweet.in_reply_to_status_id_str, {}, (error, newtweet, response) => {
+            if (!error) {
+                var promise = this.process_tweet(roomid, newtweet, depth)
+                if(promise != null){
+                  return promise.then(() => {
+                        resolve();
+                  });
+                }
+                else{
+                  resolve();
+                }
+            }
+            else
+            {
+                log.error("process_tweet: GET /statuses/show returned: " + error);
```

Comment:
Ewwww, you're going to return a promise which will **never** be fulfilled should this HTTP hit fail?! Why aren't you `reject()`ing it like you do elsewhere?